server/src/main/java/keywhiz/service/daos/SecretSeriesDAO.java

```diff
@@ -282,38 +281,27 @@ public ImmutableList<SecretSeries> getSecretSeries(@Nullable Long expireMaxTime,
       @Nullable Group group, @Nullable Long expireMinTime, @Nullable String minName,
       @Nullable Integer limit) {
 
-    Table<SecretsContentRecord> secretsContentTable = SECRETS_CONTENT;
-    if (expireMaxTime != null && expireMaxTime > 0) {
-      // Force this join to use the index on the secrets_content.expiry
-      // field. The optimizer may fail to use this index when the SELECT
-      // examines a large number of rows, causing significant performance
-      // degradation.
-      secretsContentTable = secretsContentTable.useIndexForJoin("secrets_content_expiry");
-    }
-
     SelectQuery<Record> select = dslContext
           .select(SECRETS.fields())
           .from(SECRETS)
-          .join(secretsContentTable)
-          .on(SECRETS.CURRENT.equal(SECRETS_CONTENT.ID))
           .where(SECRETS.CURRENT.isNotNull())
           .getQuery();
-    select.addOrderBy(SECRETS_CONTENT.EXPIRY.asc(), SECRETS.NAME.asc());
+    select.addOrderBy(SECRETS.EXPIRY.asc(), SECRETS.NAME.asc());
 
     // Set an upper bound on expiration dates
     if (expireMaxTime != null && expireMaxTime > 0) {
       // Set a lower bound of "now" on the expiration only if it isn't configured separately
       if (expireMinTime == null || expireMinTime == 0) {
         long now = System.currentTimeMillis() / 1000L;
-        select.addConditions(SECRETS_CONTENT.EXPIRY.greaterOrEqual(now));
+        select.addConditions(SECRETS.EXPIRY.greaterOrEqual(now));
       }
-      select.addConditions(SECRETS_CONTENT.EXPIRY.lessThan(expireMaxTime));
+      select.addConditions(SECRETS.EXPIRY.lessThan(expireMaxTime));
     }
```

Comment:
In essense, every reference to SECRETS_CONTENT.EXPIRY is now just SECRETS.EXPIRY (the value is the same), and we omit a no-longer-necessary join.

There is a very subtle difference in that, because we lack referential integrity (no foreign keys), it's possible that a secret that would have been excluded by the join (a "dangling" current secret content reference) would now be included. This should be a non-issue in practice.